sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -18,19 +18,28 @@ package com.nvidia.spark.rapids
 
 import java.io.File
 import java.util.concurrent.ConcurrentHashMap
+import java.util.function.BiFunction
+
+import scala.collection.mutable.ArrayBuffer
 
 import ai.rapids.cudf._
 import com.nvidia.spark.rapids.StorageTier.StorageTier
 import com.nvidia.spark.rapids.format.TableMeta
 
-import org.apache.spark.sql.rapids.RapidsDiskBlockManager
+import org.apache.spark.sql.rapids.{RapidsDiskBlockManager, TempSpillBufferId}
 
 /** A buffer store using GPUDirect Storage (GDS). */
 class RapidsGdsStore(
     diskBlockManager: RapidsDiskBlockManager,
     catalog: RapidsBufferCatalog = RapidsBufferCatalog.singleton)
     extends RapidsBufferStore(StorageTier.GDS, catalog) with Arm {
-  private[this] val sharedBufferFiles = new ConcurrentHashMap[RapidsBufferId, File]
+  private val blockSize = 4096
+  private val batchWriteBufferSize = 1 << 27 // 128 MiB
```

Comment:
Curious, how did we arrive at this number?  Experimentally or just guessing?  Would be nice to make it as low as possible as it's a permanent allocation.

Reply:
UCX is using BAR1 aggressively too. We normally max out the BAR1 usage by splitting it between send and receive buffers. Make sure that it isn't UCX doing the mapping (we configure it using the bounce buffer pool size).

Reply:
Ok, made this configurable and default to 8 MiB.